src/components/selectionBoxLayer.ts

```diff
@@ -62,10 +56,12 @@ export module Components {
     public bounds(newBounds: Bounds): SelectionBoxLayer;
     public bounds(newBounds?: Bounds): any {
       if (newBounds == null) {
-        return this._boxBounds;
+        return this._getBounds();
```

Comment:
As experienced with `XDragBoxLayer` / `YDragBoxLayer`, it is incredibly disruptive to have a getter error unless there is a good exception architecture surrounding the language.  In which case, I would much rather return something similar to `null` than error.